v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/FileFormatFactory.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.transforms;
+
+import com.google.auto.value.AutoValue;
+import java.util.Arrays;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.POutput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FileFormatFactory} class is a {@link PTransform} that takes in {@link PCollection} of
+ * kafka records.The Transform writes this records to GCS file(s) in user specified format.
+ */
+@AutoValue
+public abstract class FileFormatFactory
+        extends PTransform<PCollection<KV<String, String>>, POutput> {
+
+    /* Logger for class. */
+    private static final Logger LOG = LoggerFactory.getLogger(FileFormatFactory.class);
+
+    public static WriteToGCSBuilder newBuilder() {
+        return new AutoValue_FileFormatFactory.Builder();
+    }
+
+    public abstract KafkaToGCSOptions options();
+
+    public abstract String outputFileFormat();
+
+    @Override
+    public POutput expand(PCollection<KV<String, String>> records) {
+        POutput output = null;
+        WriteToGCSUtility.FileFormat outputFileFormat;
+        try {
+            outputFileFormat = WriteToGCSUtility.FileFormat.valueOf(outputFileFormat());
```

Comment:
If this and the main class are the only two places where FileFormat is used, then perhaps we should move it here instead of having it tucked away in WriteToGCSUtility?

Reply:
FileFormat is also used in WriteToxxx classes for setting prefix. And moved FileFormatFactory to kafkatogcs module. If I move FileFormat in FileFormatFactory it will create a cyclic dependency.

Reply:
Thats fine, we can address this later if needed.